cmd/ostracon/commands/show_validator_test.go

```diff
@@ -0,0 +1,203 @@
+	require.NoError(t, err)
+	init := NewInitCmd()
+	err = init.RunE(init, nil)
+	require.NoError(t, err)
+	output, err := captureStdout(func() {
+		err = ShowValidatorCmd.RunE(ShowValidatorCmd, nil)
+		require.NoError(t, err)
+	})
+	require.NoError(t, err)
+
+	// output must match the locally stored priv_validator key
+	privKey := loadFilePVKey(t, config.PrivValidatorKeyFile())
+	bz, err := tmjson.Marshal(privKey.PubKey)
+	require.NoError(t, err)
+	require.Equal(t, string(bz), output)
+}
+
+func TestShowValidatorWithoutLocalKeyFile(t *testing.T) {
+	setupEnv(t)
+	config := cfg.DefaultConfig()
+	if tmos.FileExists(config.PrivValidatorKeyFile()) {
+		err := os.Remove(config.PrivValidatorKeyFile())
+		require.NoError(t, err)
+	}
+	err := showValidator(ShowValidatorCmd, nil, config)
+	require.Error(t, err)
+}
+
+func TestShowValidatorWithKMS(t *testing.T) {
+	dir := setupEnv(t)
+	cfg.EnsureRoot(dir)
+
+	original := config
+	defer func() {
+		config = original
+	}()
+
+	config = cfg.DefaultConfig()
+	config.SetRoot(dir)
+	err := RootCmd.PersistentPreRunE(RootCmd, nil)
+	require.NoError(t, err)
+	init := NewInitCmd()
+	err = init.RunE(init, nil)
+	require.NoError(t, err)
+
+	chainID, err := loadChainID(config)
+	require.NoError(t, err)
+
+	if tmos.FileExists(config.PrivValidatorKeyFile()) {
+		err := os.Remove(config.PrivValidatorKeyFile())
+		require.NoError(t, err)
+	}
+	privval.WithMockKMS(t, dir, chainID, func(addr string, privKey crypto.PrivKey) {
+		config.PrivValidatorListenAddr = addr
+		require.NoFileExists(t, config.PrivValidatorKeyFile())
+		output, err := captureStdout(func() {
+			err := showValidator(ShowValidatorCmd, nil, config)
+			require.NoError(t, err)
+		})
+		require.NoError(t, err)
+
+		// output must contains the KMS public key
+		bz, err := tmjson.Marshal(privKey.PubKey())
+		require.NoError(t, err)
+		expected := string(bz)
+		require.Contains(t, output, expected)
+	})
+}
+
+func TestShowValidatorWithInefficientKMSAddress(t *testing.T) {
+	dir := setupEnv(t)
+	cfg.EnsureRoot(dir)
+
+	original := config
+	defer func() {
+		config = original
+	}()
+
+	config = cfg.DefaultConfig()
+	config.SetRoot(dir)
+	err := RootCmd.PersistentPreRunE(RootCmd, nil)
+	require.NoError(t, err)
+	init := NewInitCmd()
+	err = init.RunE(init, nil)
+	require.NoError(t, err)
+
+	if tmos.FileExists(config.PrivValidatorKeyFile()) {
+		err := os.Remove(config.PrivValidatorKeyFile())
+		require.NoError(t, err)
+	}
+	config.PrivValidatorListenAddr = "127.0.0.1:inefficient"
+	err = showValidator(ShowValidatorCmd, nil, config)
+	require.Error(t, err)
+}
+
+func TestLoadChainID(t *testing.T) {
+	dir := setupEnv(t)
+	cfg.EnsureRoot(dir)
+
+	original := config
+	defer func() {
+		config = original
+	}()
+
+	config = cfg.DefaultConfig()
+	config.SetRoot(dir)
+	err := RootCmd.PersistentPreRunE(RootCmd, nil)
+	require.NoError(t, err)
+	init := NewInitCmd()
+	err = init.RunE(init, nil)
+	require.NoError(t, err)
+	require.FileExists(t, config.GenesisFile())
```

Comment:
Does `loadChainID` depend on `RootCmd.PersistentPreRunE` and `init.RunE`?

I think `loadChainID` is a local function and it doesn't need to depend on some public commands.

If testing `loadChainID`, I think the below is uncessarry.
```
	config = cfg.DefaultConfig()
	config.SetRoot(dir)
	err := RootCmd.PersistentPreRunE(RootCmd, nil)
	require.NoError(t, err)
	init := NewInitCmd()
	err = init.RunE(init, nil)
	require.NoError(t, err)
	require.FileExists(t, config.GenesisFile())
```

Reply:
In order to use `loadChainID()`, we need an environment where genesis.json and state db exist. To create them formally, I thought it would be reliable and easy to create an initialized directory with the same action as `init`.

On the other hand, there are several ways to configure the initial directory for testing, and  `cfg.ResetTestRootWithChainID()` is one of them. I don't have a strong preference for the way the directory is initially configured for the test, so I'll modify this to do it that way.